rust/lance/src/encodings/binary.rs

```diff
@@ -182,6 +182,25 @@ impl<'a, T: ByteArrayType> BinaryDecoder<'a, T> {
         Ok(Arc::new(as_primitive_array(&values).clone()))
     }
 
+    fn count_nulls<O: OffsetSizeTrait>(offsets: &ScalarBuffer<O>) -> (usize, Option<Buffer>) {
+        let mut null_count = 0;
+        let mut null_buf = MutableBuffer::new_null(offsets.len() - 1);
+        offsets.windows(2).enumerate().for_each(|(idx, w)| {
+            if w[0] == w[1] {
```

Comment:
Can an empty string be distinguished from a null?

Reply:
No, it cannot. That's a limitation of our format.

Reply:
Let's fix this via the null support :)